cli/src/commands/cast/mod.rs

```diff
@@ -74,46 +74,56 @@ pub fn command(matches: Option<&ArgMatches>) {
             let function_artifact_path =
                 format!("./net/services/{}/{}", service_name, function_name);
 
-            let wasm_path = lang::compile(project.clone(), &service_name, function);
-
-            // TODO zip not needed w/ container functions
-            let mut function_dirs = vec![wasm_path];
-            if let Some("ruby") = function.language.clone().as_deref() {
-                function_dirs.push(PathBuf::from(format!(
-                    "{}/rubysrc",
-                    &function_artifact_path
-                )));
-            }
-            archive::zip_dirs(
-                function_dirs,
-                format!("{}/{}.zip", function_artifact_path.clone(), &function_name),
-                Vec::new(),
-            )
-            .expect("unable to zip function artifacts");
-        }
-    }
-
-    {
-        let ctx = Rc::new(
-            Context::from_project(project.clone(), asml_manifest)
-                .expect("could not make context from manifest"),
-        );
-        let artifacts = ctx
-            .cast(ctx.clone(), None)
-            .expect("could not cast assemblylift context");
-        for artifact in artifacts {
-            let path = artifact.write_path;
-            let mut file = match fs::File::create(path.clone()) {
-                Err(why) => panic!("couldn't create file {}: {}", path.clone(), why.to_string()),
-                Ok(file) => file,
+            match lang::compile(project.clone(), &service_name, function).unwrap() {
+                wasm_path => {
+                    let mut function_dirs = vec![wasm_path.clone()];
+                    if let Some("ruby") = function.language.clone().as_deref() {
+                        function_dirs.push(PathBuf::from(format!(
+                            "{}/rubysrc",
+                            &function_artifact_path
+                        )));
+                    }
+
+                    // TODO zip not needed w/ container functions
+                    archive::zip_dirs(
+                        function_dirs,
+                        format!("{}/{}.zip", function_artifact_path.clone(), &function_name),
+                        Vec::new(),
+                    )
+                        .expect("unable to zip function artifacts");
+
+                    {
+                        let ctx = Rc::new(
+                            Context::from_project(project.clone(), asml_manifest.clone())
+                                .expect("could not make context from manifest"),
+                        );
+                        let artifacts = ctx
+                            .cast(ctx.clone(), None)
+                            .expect("could not cast assemblylift context");
+
+                        // todo: serialize to string only if the file is not out of date
```

Comment:
I think we can use the sha2 crate for this to compare the sha256 of the existing file contents and the new artifact

https://github.com/RustCrypto/hashes

Reply:
heyy. i didnt realize you already replied/reviewed. plus ive been mad busy jajajaja. i'll take a look, update and revert here again.